ddtrace/pin.py

```diff
@@ -1,42 +1,54 @@
 import logging
-import wrapt
 
+import wrapt
 import ddtrace
 
+
 log = logging.getLogger(__name__)
 
-_DD_PIN_NAME = '_datadog_pin'
 
 # To set attributes on wrapt proxy objects use this prefix:
 # http://wrapt.readthedocs.io/en/latest/wrappers.html
+_DD_PIN_NAME = '_datadog_pin'
 _DD_PIN_PROXY_NAME = '_self_' + _DD_PIN_NAME
 
 
 class Pin(object):
-    """ Pin (a.k.a Patch INfo) is a small class which is used to
-        set tracing metadata on a particular traced connection.
-        This is useful if you wanted to, say, trace two different
-        database clusters.
+    """Pin (a.k.a Patch INfo) is a small class which is used to
+    set tracing metadata on a particular traced connection.
+    This is useful if you wanted to, say, trace two different
+    database clusters.
 
         >>> conn = sqlite.connect("/tmp/user.db")
         >>> # Override a pin for a specific connection
         >>> pin = Pin.override(conn, service="user-db")
         >>> conn = sqlite.connect("/tmp/image.db")
     """
+    __slots__ = ['app', 'app_type', 'tags', 'tracer', '_target', '_config', '_initialized']
 
-    __slots__ = ['app', 'app_type', 'service', 'tags', 'tracer', '_initialized']
-
-    def __init__(self, service, app=None, app_type=None, tags=None, tracer=None):
+    def __init__(self, service, app=None, app_type=None, tags=None, tracer=None, _config=None):
         tracer = tracer or ddtrace.tracer
-        self.service = service
         self.app = app
         self.app_type = app_type
         self.tags = tags
         self.tracer = tracer
+        self._target = None
+        # keep the configuration attribute internal because the
+        # public API to access it is not the Pin class
+        self._config = _config or {}
```

Comment:
We don't need to copy it here, since any operation to use the pin, calls the `clone()` method once. Indeed, it's possible that we:
* initialize the global config with our defaults
* first set the `Pin` in one of our integrations
* users update the configuration
* when the `Pin` is used in the integration for the first time (via `get_from`) it gets cloned

If we copy the dictionary, we'll have always our defaults.